code/modules/modular_computers/hardware/hard_drive.dm

```diff
@@ -156,6 +156,8 @@
 /obj/item/computer_hardware/hard_drive/Destroy()
 	if(parent_computer?.hard_drive == src)
 		parent_computer.hard_drive = null
+	for(var/datum/computer_file/file in stored_files)
+		file.hard_drive = null
```

Comment:
This isn't correct, if the hard drive is deleted, the programs it contains should be cleanly deleted too, not left hanging unlinked